solutions.Rmd

```diff
@@ -133,3 +154,142 @@ ggplot(data = mpg, aes(x=displ, y=hwy))+
+
+ggplot(data = mpg) + geom_bar(mapping = aes(x = class, y = displ, fill = drv), stat = "summary", 
+                              fun.y = "mean", position = "dodge")
+
+```
+
+---
+
+Extension: Manually change the colours of the fill being used. Also change the font size and angle of the x axis text labels so that these can be read in the viewer.
+
+
+```{r, fig.height = 3}
+
+ggplot(data = mpg) + 
+  geom_bar(mapping = aes(x = class, y = displ, fill = drv), stat = "summary", 
+           fun.y = "mean", position = "dodge") +
+  theme(axis.text.x = element_text(angle = 90, size = 8)) + 
+  scale_fill_manual(values = c("blue", "darkblue", "blueviolet"))
+
+```
+
+## 12
+
+Create a basic scatter plot showing highway fuel efficiency vs city fuel efficiency. Add a Wall Street Journal theme from the ggthemes package; and colour the points based on the number of cylinders the car has.
+
+Change the title and axis labels to something sensible based on what the chart is showing.
+
+```{r, fig.height = 3}
+
+ggplot(data = mpg) +
+  geom_point(mapping = aes(x = hwy, y = cty, colour = as.factor(cyl))) + 
+  theme_wsj() +
+  labs(title = "Fuel efficiency comparison between city and highway", 
+       x = "Highway efficiency", y = "City efficiency")
+
+```
+
+---
+
+Create a custom theme changing: major gridlines to blue; x axis font colour to red and y axis font colour to purple; move the legend position to the bottom of the plot.
+
+```{r, fig.height = 3}
+
+ggplot(data = mpg) +
+  geom_point(mapping = aes(x = hwy, y = cty, colour = as.factor(cyl))) + 
+  labs(title = "Fuel efficiency comparison between city and highway", 
+       x = "Highway efficiency", 
+       y = "City efficiency") +
+  theme(panel.grid.major = element_line(colour = "blue"),
+        axis.text.x = element_text(colour = "red"),
+        axis.text.y = element_text(colour = "purple"),
+        legend.position = "bottom")
+
+```
+
+---
+
+Try changing the x and y axis so that the scale starts at zero. 
+
+Change the title of the legend to 'Number of Cylinders' and re-name the labels with the text version of the number (e.g. 'four' instead of '4')
+
+```{r, fig.height = 3}
+
+ggplot(data = mpg) +
+  geom_point(mapping = aes(x = hwy, y = cty, colour = as.factor(cyl))) + 
+  labs(title = "Fuel efficiency comparison between city and highway", 
+       x = "Highway efficiency", 
+       y = "City efficiency") +
+  xlim(0,50) + ylim(0,40) +
+  scale_colour_discrete(labels = c("Four", "Five", "Six", "Eight"), name = "Number of Cylinders")
+
+```
+
+## 13
+
+Plot a line chart to show change in city efficiency between the two years, with a different line colour and line type for each class. Make sure you specify the dataset as mpg_group. 
+
+Use the viridis scale to colour the lines for a black and white friendly printed version.
+
+```{r, fig.height = 3}
+
+mpg_group    <- mpg %>% select(class, year, cty) %>% group_by(class, year) %>% summarise_all(mean)
+
+ggplot(data = mpg_group) + 
```

Comment:
This plot is pretty hard to read as-is. The chart with the thicker lines on the next slide is an improvement, and perhaps that's the point, but this could definitely be improved.

Reply:
I think the problem is I'm trying to create a line chart example using the mpg dataset, which doesn't have variables that are suited to this (hence the data manipulation before plotting...).
I've made the solution chart bigger by moving on to its own slide; and fixed the x and y axis labels to make it a bit more readable. Open to any other suggestions though for how to make this a better example of a line chart